File: DAL/Entities/Doctor.cs
﻿using DAL.Entities.Entity;
using System;
using System.Collections.Generic;
using System.Text;

namespace DAL.Entities
{
    public class Doctor : Entity<int>
    {
        public string LastName { get; set; }
        public string FirstName { get; set; }
        public string Patronymic { get; set; }
        public string Specialty { get; set; }
        public string Address { get; set; }
        public string PhoneNumber { get; set; } 
        public int Cabinet { get; set; }

        public IEnumerable<DoctorSchedule> Schedules { get; set; }
        public IEnumerable<Visit> Visits { get; set; }
        public IEnumerable<Appointment> Appointments { get; set; }
    }
}

File: PL/ClientApp/src/app/_models/doctor.ts
import { Appointment } from "./appointment";
import { DoctorSchedule } from "./doctorSchedule";
import { Visit } from "./visit";

export class Doctor {
  id: number;
  lastName: string;
  firstName: string;
  patronymic: string;
  specialty: string;
  address: string;
  phoneNumber: string;
  cabinet: number;

  schedules: DoctorSchedule[];
  appointments: Appointment[];
  visits: Visit[];
}
